### ISSUE TITLE
Rounding is not clearly documented

### ISSUE THREAD

**Issue by jan-osch** [ericelliott/moneysafe#46]
I am looking for a safe operation to change money amounts between different currencies. 

```js
const { $, multiply } = require('moneysafe');
const multiplyMoney = (first, second) => multiply($.of(first), $.of(second)).valueOf();

const amount = 20000
const rate = 3.663

multiplyMoney(amount, rate) //  73200  
amount * rate // 73260
```

Any idea why I get the rounding and how to avoid it?

**Comment by WilliamColmenares** [comment]
Same issue here

**Comment by mayankkamboj47** [comment]
**Why it's happening : **

BigNumber dependency uses the .toString() function to convert what we have in dollars to BigNumber. 
For example : 
`let money = $(5.553);
 console.log(new BigNumber(money));
 /* -> BigNumber { s: 1, e: 0, c: [ 5, 55000000000000 ] }
 money.toString = ()=>'5.553';
 console.log(new BigNumber(money));
 //-> BigNumber { s: 1, e: 0, c: [ 5, 55300000000000 ] }
`

In your situation, the 3.663 is being truncated to 3.66 which gives 73200 dollars instead of 73260.

** How to get around it :**

One way I can think of is to create your own different currency,
`
const new$ = createCurrency({decimal:3});
`
And work on this. This solves the problem in your case, but should you then need to multiply with a number that has 4 decimal places, we run into the same issue. 

A better and long term solution, is that we need a way not of multiplying two currencies together, but of multiplying a currency with a number. For instance, if we're taking forty dollars 10 times, it makes more sense to do $40*10 than to do : $40*$10. The latter should, produce 400 "Dollars squared", which doesn't make much sense.

We can change the times function from `const times = b => of(value.multipliedBy(of(b)));
` to `const times = b => of(value.multipliedBy(b));`. This will allow you to multiply a currency with a number without getting the rounding on the number (you'll still get rounding on the currency, which is intentional behaviour).

**Comment by ericelliott** [comment]
This should be fixed in v2.2.2
